src/nisarqa/processing/insar/quiver_plots.py

```diff
@@ -382,4 +683,261 @@ def truncate_colormap(
+    """
+    Parameters to convert geocoded offsets values from az/range to projected.
+
+    For e.g. GOFF and GUNW, the along track offsets and slant range offsets
+    rasters are geocoded to projected coordinates, but their pixels'
+    values represent offsets in azimuth and slant range directions (in meters).
+    These parameters are needed to rotate/scale the pixel values from
+    azimuth and slant range directions into the X/Y directions of the projected
+    coordinates.
+
+    Parameters
+    ----------
+    orbit : isce3.core.Orbit
+        The trajectory of the radar antenna phase center over a time interval
+        that includes the observation times of each point offset pixel.
+    look_side : isce3.core.LookSide or {'left', 'right'}
+        The look direction of the input product (left-looking or right-looking).
+    wavelength : float
+        The radar central wavelength for the frequency subband of the offset,
+        in meters.
+    geo2rdr_params : dict of [str, float] or None, optional
+        An optional dict of parameters configuring the behavior of the
+        root-finding routine used in geo2rdr (bracketing implementation).
+        The following keys are supported:
+        'tol_aztime':
+           Azimuth time convergence tolerance, in seconds. Defaults to 1e-7.
+        'time_start':
+           Start of search interval, in seconds. Defaults to `orbit.start_time`.
+        'time_end':
+           End of search interval, in seconds. Defaults to `orbit.end_time`.
+        Defaults to None.
+    """
+
+    orbit: isce3.core.Orbit
+    wavelength: float
+    look_side: isce3.core.LookSide | str
+    geo2rdr_params: Mapping[str, float] | None = None
+
+
+def get_offset_values_in_projected_coordinates(
+    az_offsets: np.ndarray,
+    rg_offsets: np.ndarray,
+    geo_grid: nisarqa.GeoGrid,
+    projection_params: nisarqa.ParamsForAzRgOffsetsToProjected,
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Convert pixel values from offset in az/range to offset in projected X/Y.
+
+    For the along-track and slant range offsets raster in NISAR GOFF and GUNW
+    products, the raster images are geocoded to projected coordinates but
+    their pixel values represent offsets in azimuth and slant range directions
+    (respectively) in meters.
+    This function will convert the values from offset amount in azimuth
+    and slant range directions to the offset in the X and Y directions
+    of the projected image grid.
+
+    Parameters
+    ----------
+    rg_offsets : numpy.ndarray
+        2D array of the values of the slant range offsets.
+        Array should be geocoded, but the pixel values should represent
+        offset in slant range direction (in meters).
+    az_offsets : numpy.ndarray
+        2D array of the values of the along track offsets.
+        Array should be geocoded, but the pixel values should represent
+        offset in along azimuth direction (in meters).
```

Comment:
Swap the order of these in the docstring to match their order in the parameter list

```suggestion
    az_offsets : numpy.ndarray
        2D array of the values of the along track offsets.
        Array should be geocoded, but the pixel values should represent
        offset in along azimuth direction (in meters).
    rg_offsets : numpy.ndarray
        2D array of the values of the slant range offsets.
        Array should be geocoded, but the pixel values should represent
        offset in slant range direction (in meters).
```